diff -Nur wget-1.13.4.org//src/progress.c wget-1.13.4/src/progress.c
--- wget-1.13.4.org//src/progress.c	2015-11-05 09:12:57.558246759 -0500
+++ wget-1.13.4/src/progress.c	2015-11-05 09:14:02.448246745 -0500
@@ -885,6 +885,7 @@
   */
   int dlbytes_size = 1 + MAX (size_grouped_len, 11);
   int progress_size = bp->width - (4 + 2 + dlbytes_size + 8 + 14);
+  int dl_per = 0;
 
   /* The difference between the number of bytes used,
      and the number of columns used. */
@@ -897,6 +898,7 @@
   if (bp->total_length > 0)
     {
       int percentage = 100.0 * size / bp->total_length;
+      dl_per = percentage;
       assert (percentage <= 100);
 
       if (percentage < 100)
@@ -992,6 +994,7 @@
       wgint dlquant = hist->total_bytes + bp->recent_bytes;
       double dltime = hist->total_time + (dl_total_time - bp->recent_start);
       double dlspeed = calc_rate (dlquant, dltime, &units);
+      print_stat("(DOWNLOADING %s %4.*f%s %d%%)",number_to_static_string(bp->total_length), dlspeed >= 99.95 ? 0 : dlspeed >= 9.995 ? 1 : 2, dlspeed, short_units[units],dl_per);	
       sprintf (p, " %4.*f%s", dlspeed >= 99.95 ? 0 : dlspeed >= 9.995 ? 1 : 2,
                dlspeed, short_units[units]);
       move_to_end (p);
